### ISSUE TITLE
Rule Request: Float-type literal decimal

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2752]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

> 1. Why should this rule be added? Share links to existing discussion about what the community thinks about this.

I've been reading a lot of open source Swift code lately and I see both `let someFloat = 1` and `let someFloat = 1.0` being used. Would be nice to enforce consistency here, either way.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

If you have set it to enforce `.0`:

Would trigger:

```swift
let foo = 3

let bar: CGFloat = 0
```

Would not trigger:

```swift
let foo = 3.0

let bar: CGFloat = 0.0
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

Yes, it should be configurable whether to enforce or disallow the `.0` part in float-type literals.

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in

**Comment by sindresorhus** [comment]
Please keep it open.

**Comment by sindresorhus** [comment]
Please keep it open.
